presto 0.150 (new formula)

ankane opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----

First off, thanks for all the great work on Homebrew!

This adds Facebook's [Presto](https://prestodb.io/). I renamed the bin file `presto-server` (from `launcher`) to keep `presto` available for the [CLI](https://prestodb.io/docs/current/installation/cli.html), but happy to change if needed.

Official Presto [setup instructions here](https://prestodb.io/docs/current/installation/deployment.html)

DomT4 commented on Formula/presto.rb:
```diff
@@ -0,0 +1,82 @@
+  def install
+    libexec.install Dir["*"]
+
+    mkdir "#{libexec}/etc"
```
Can drop this line. It'll be created by the `.write` below automagically.

DomT4 commented on Formula/presto.rb:
```diff
@@ -0,0 +1,82 @@
+  end
+
+  test do
+    system "#{bin}/presto-server", "run", "--help"
```
Minor nit, but we _slightly_ prefer:
```ruby
system bin/"presto-server", "run", "--help"
```
when used in this way.

DomT4 commented on Formula/presto.rb:
```diff
@@ -0,0 +1,82 @@
+
+    (libexec/"etc/node.properties").write <<-EOS.undent
+      node.environment=production
+      node.id=ffffffff-ffff-ffff-ffff-ffffffffffff
```
Is this something users will need to edit or will it be fine like this?

ankane replied:
Yeah, no need to edit it.

ankane commented:
Sorry for all the noise.  I updated it based on your feedback and added the `presto` client, a default connector, and instructions for adding new connectors as well.

DomT4 commented on Formula/presto.rb:
```diff
@@ -0,0 +1,99 @@
+      exec "#{libexec}/bin/launcher" "$@"
+    EOS
+
+    resource("presto-cli").stage do |s|
```
Should be able to do something like:
```ruby
resource("presto-cli").stage do
  bin.install "presto-cli-#{version}-executable.jar" => "presto"
end
```
Maybe?

ankane replied:
Looks like it works. Updated.

DomT4 commented on Formula/presto.rb:
```diff
@@ -0,0 +1,99 @@
+    (libexec/"etc/node.properties").write <<-EOS.undent
+      node.environment=production
+      node.id=ffffffff-ffff-ffff-ffff-ffffffffffff
+      node.data-dir=#{var}/presto/data
```
Can you add a `post_install` block to `mkpath` this directory? Lots of examples if you search for `post_install` through the formula if necessary.

DomT4 commented on Formula/presto.rb:
```diff
@@ -0,0 +1,99 @@
+      discovery.uri=http://localhost:8080
+    EOS
+
+    (libexec/"etc/log.properties").write <<-EOS.undent
```
If this goes under ~80 characters, which I think it _just_ about does, don't need the `EOS.undent` element.
```ruby
(libexec/"etc/log.properties").write "com.facebook.presto=INFO"
```
etc. We try to use the blocks only for multiple lines & not stretching formulae out too widely.

ankane commented:
Updated based on feedback

MikeMcQuaid commented on Formula/presto.rb:
```diff
@@ -0,0 +1,99 @@
+  def caveats; <<-EOS.undent
+    Add connectors to #{libexec}/etc/catalog/. See:
+    https://prestodb.io/docs/current/connector.html
+    EOS
```
Remove these caveats, they aren't Homebrew-specific information, thanks!

ankane replied:
Hey @MikeMcQuaid, the Presto set up instructions mention adding connectors relative to the installation.  However, I'm not sure most people know exactly where Homebrew installs packages (they problably know `/usr/local`, but not further down the tree), so I think this could save people a lot of time. Is there a better place to add this information?

MikeMcQuaid replied:
That's cool, will leave as-is then, thanks.

MikeMcQuaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

ankane commented:
Thanks for the feedback and merging. The work your team does is great. 👍
